sous-chef/sous_chef/formatter/format_unit.py

```diff
@@ -25,50 +25,46 @@ def __str__(self):
 
 @dataclass
 class UnitFormatter:
+    @staticmethod
     def convert_to_desired_unit(
-        self, quantity: float, pint_unit: Unit, desired_pint_unit: Unit
-    ) -> (float, str, Unit):
+        quantity: float, pint_unit: Unit, desired_pint_unit: Unit
+    ) -> (float, Unit):
         original_value = quantity * pint_unit
         converted_value = original_value.to(desired_pint_unit)
-
         # round to significant digits per defined unit_registry
         converted_quantity = round(converted_value.magnitude, 2)
-        converted_pint_unit = converted_value.units
-        converted_unit = self._get_unit_as_abbreviated_str(
-            converted_value.units
-        )
-        return converted_quantity, converted_unit, converted_pint_unit
+        return converted_quantity, converted_value.units
 
-    def extract_unit_from_text(self, text_unit: str) -> (str, Unit):
-        pint_unit = self._get_pint_unit(text_unit)
-        true_unit = self._get_unit_as_abbreviated_str(pint_unit)
-        return true_unit, pint_unit
+    @staticmethod
+    def get_unit_str(quantity: float, pint_unit: Unit) -> str:
+        if pint_unit == unit_registry.dimensionless:
+            return ""
 
-    def get_unit_str(self, quantity: float, unit: Unit) -> str:
-        # TODO more robust way to do? e.g. inflection or define
-        if quantity > 1 and unit in not_abbreviated:
-            return f"{unit}s"
-        return self._get_unit_as_abbreviated_str(unit)
+        if pint_unit in not_abbreviated:
+            # TODO more robust way to do? e.g. inflection or define
+            if quantity > 1:
+                return f"{pint_unit}s"
 
-    @staticmethod
-    def _get_pint_unit(text_unit: str):
-        try:
-            pint_unit = unit_registry.parse_expression(
-                text_unit.casefold()
-            ).units
-            if pint_unit in allowed_unit_list:
-                return pint_unit
-            FILE_LOGGER.warning(
-                "[get pint unit]",
-                warn="unit not in allowed_unit_list",
-                unit=pint_unit,
-            )
-        except (AttributeError, DimensionalityError, UndefinedUnitError):
-            pass
-        raise UnitExtractionError(text=text_unit)
+            if pint_unit in not_abbreviated:
+                return str(pint_unit)
+
+        return "{:~}".format(pint_unit)
 
     @staticmethod
-    def _get_unit_as_abbreviated_str(unit: Unit) -> str:
-        if unit in not_abbreviated:
-            return str(unit)
-        return "{:~}".format(unit)
+    def get_pint_unit(text_unit: str) -> Unit:
```

Comment:
minor: `get_pint_unit` and `get_pint_repr` - do we need separation?

Reply:
Yes, this is a bit nicer decoupled as `get_pint_repr` is used in another place, and `get_pint_unit` appears in a lot of tests as part of `UnitFormatter` instance.